clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AWSServerGroupNameResolver.groovy

```diff
@@ -64,7 +64,7 @@ class AWSServerGroupNameResolver extends AbstractServerGroupNameResolver {
       return []
     }
 
-    def serverGroupsInRegion = cluster.serverGroups.findAll { it.region == region }
+    def serverGroupsInRegion = cluster.serverGroups.findAll { it.region == region && asgService.getAutoScalingGroup(it.name) }
```

Comment:
you are going to burn a bunch of API calls (`asgService` goes directly against AWS).